plugins/manual_maps/ManualMapLoading.ts

```diff
@@ -0,0 +1,214 @@
+import fs from 'fs/promises'
+
+import { Logger } from '../../src/Logger.js'
+import { Client } from '../../src/client/Client.js'
+import config from '../../config/Config.js'
+import { MapService } from "../../src/services/MapService.js"
+
+type ReturnCode = {
+  code: number
+}
+
+export class ManualMapLoading {
+  static readonly prefix: string = config.mapsDirectoryPrefix
+  static mapIndex: number = 0
+  static oldQueue: tm.Map[]
+  static oldCurr: tm.CurrentMap
+
+  /**
+   * Parse every map in the `config.mapsDirectoryPrefix/config.mapsDirectory
+   * @param presentMaps list of already existing maps to compare with
+   * @param dirname the directory name
+   * @returns list of maps in the same format that the server returns them
+   */
+  static async parseMaps(presentMaps: tm.Map[] = [], dirname: string = config.mapsDirectory) {
+    const filesOrDirs = await fs.readdir(this.prefix + dirname, {withFileTypes: true})
+    if (filesOrDirs.length > 5000) Logger.warn(`Trying to parse a large amount of maps (${filesOrDirs.length}), reading their info might take a while.`)
+    const maps: tm.Map[] = []
+    const addedMaps: tm.Map[] = []
+    const removedMaps: tm.Map[] = []
+    for (const f of filesOrDirs) {
+      if (f.isDirectory()) {
+        (await this.parseMaps(presentMaps, dirname + f.name)).forEach(a => {
+          maps.push(a[0])
+          addedMaps.push(a[1])
+          removedMaps.push(a[2])
+        })
+        continue
+      }
+      const map = await this.parseMap(dirname + '/' + f.name, presentMaps, maps)
+      if ((<ReturnCode>map).code === 2) continue // duplicate file
```

Comment:
i have several questions regarding this syntax, just use ```as```

Reply:
This was what intellij recommended I do, Ill try with as.